### ISSUE TITLE
Combined gradients

### ISSUE THREAD

**Issue by ianhbell** [autodiff/autodiff#293]
I have a function like so:
``
f(T,rho,x)
``
in which T and rho are scalars and x is a vector. I would like to calculate the combined gradient

$$
\left(\frac{\partial^2 f}{\partial T \partial \vec x}\right)_{\rho}
$$

with autodiff. Is there such a functionality short of looping over the values of x? It isn't exactly a normal gradient.

**Comment by allanleal** [comment]
Hi Ian, unfortunately there is no out-of-the-box function to produce these second-order cross derivatives. 

I assume you are using at least a second-order dual number. What I suggest you to do is to play with the seeding of the $T$ and $x$ variables when you evaluate $f$ so that you can produce $(\partial^2 f/\partial T \partial x_i)_{x_j\neq x_i,\rho}$ in a single evaluation of $f$. By seeding I mean setting `T.grad = 1.0` and `x[i].grad.grad = 1`. You'll still need to loop through $x_i$ though. It's been a while that I don't play with dual numbers (I'm mainly using `real` because I don't need cross derivatives) so I can't say what will be the outcome of this experiment.

**Comment by ianhbell** [comment]
Yes I have mostly switched to ``real`` as well; I had a place where I was taking 9th derivatives with duals, and the duals were causing compilation to crash or at least be very slow, but the reals caused no problem. 

I have a working solution with multiple function calls, but I am sure this is not the most ideal so I will revisit this sorcery later on when I get back into profiling.
